subtitle.pm

```diff
@@ -0,0 +1,70 @@
+# 2015 by Bernhard M. Wiedemann
+# Licensed under GPL v2 or later
+
+package subtitle;
+use threads;
+use threads::shared;
+
+our $fps         = 24;
+our $granularity = int($fps * 0.5);
+our $ssnoffset   = 0;
+our $subopts     = " line:0 align:start";
+
+sub ssn2videotime($) {
+    my $ssn = shift;
+    my $sec = ($ssn + $ssnoffset) / $fps;
+    return sprintf("%02i:%02i.%03i", $sec / 60, $sec % 60, 1000 * ($sec - int($sec)));
+}
+
+our $subnum;
+our $subtext : shared;
+our $starttime;
+our $subfd;
+
+sub init_subtitle($) {
+    my $filename = shift;
+    open($subfd, ">", $filename) or die $!;
+    print $subfd "WEBVTT FILE\n\n";
+    $subtext   = '';
+    $subnum    = 1;
+    $starttime = undef;
+}
+
+sub finish_subtitle() {
+    close $subfd;
+    $subfd = undef;
+}
+
+sub add_subtitle_line($) {
+    return unless $subfd;
+    local $_ = shift;
+    if (m/wrote screenshot #(\d+)/) {
+        my $ssn = $1;
```

Comment:
this is really bizzare code. Passing strings to functions who then parse integers out of it.